server.go

```diff
@@ -142,6 +144,20 @@ func (s *Server) initialize(interceptorChain *rpcperms.InterceptorChain) error {
 		return fmt.Errorf("unable to start asset custodian: %v", err)
 	}
 
+	// We need to fetch all unspent assets from the database so the re-org
+	// watcher can properly detect re-orgs for not yet fully buried assets.
+	ctx := context.Background()
+	chainAssets, err := s.cfg.AssetStore.FetchAllAssets(ctx, false, nil)
```

Comment:
I think we only need to read all the assets not sufficiently buried? So we can  use a depth filter here to check what the height of their current confirmed managed UTXO is. This could help us consderiably cut down on start up time as the amount of assets the daemon holds scales up.